test/parc/parc16.kk

```diff
@@ -1,7 +1,7 @@
 fun test(xs:list<int>, y:int, z:int) {
   // drop z
   match(xs) {
-    Cons(x,_) ->
+    Cons(x as 0) ->
```

Comment:
This change looks surprising to me: where did the second argument of `Cons` go? And why do you match `x` against `0` now?

In any case, I would prefer not to change the existing tests, but rather create a new test.

Reply:
At one point the optimization also removed unreachable cases like the second case here which will never be reached due to the first case matching no matter what. So the output changed in a way that was unrelated to what is tested here. I added the 0 to ensure the second case would be reachable.

However, I think I reverted when it removed unreachable cases, because I wasn't thoroughly checking guards. So let me revert the change in the test, and see how the output changes.

Reply:
Made a fix, now the test output doesn't change.